New/NEW Product Variants API, add variant stock to response

kastoras opened the pull request:
When acessing product variants via GET API /products/{id}/variants, Include stock_warehouse data to the response

kastoras committed:
New/NEW Product Variants API, add variant stock to response

When acessing product variants via GET API /products/{id}/variants, Include stock_warehouse data to the response

kastoras committed:
New/NEW Product Variants API, add variant stock to response

When acessing product variants via GET API /products/{id}/variants, Include stock_warehouse data to the response

stickler-ci committed:
Fixing style errors.

eldy commented on htdocs/product/class/api_products.class.php:
```diff
@@ -1574,6 +1575,11 @@ public function getVariants($id)
+
+			$productModel = new Product($this->db);
+			$productModel->fetch((int) $combination->fk_product_child);
+			$productModel->load_stock();
```
Because we want API to be a low level and this method may generate a high level of subrequest, it is better to add an option to do this.
For example, other api have a parameter call $includestockdata set to 0 by default. And load of stock is done only if this param is set to 1. Can you add this ?

kastoras replied:
Yes, of course. I will make that change.

FHenry committed:
Add ORDER CREATE trigger for notificiation

eldy committed:
Merge pull request #17423 from FHenry/dev_new_notif_trigger

Add ORDER_CREATE trigger for notificiation

kastoras committed:
Merge pull request #1 from Dolibarr/develop

Get latest changes

kastoras committed:
New/NEW Product Variants API, add variant stock to response by parameter

If $includestock parameter is set to 1, the response will contain variant stock data

stickler-ci committed:
Fixing style errors.

kastoras committed:
Merge remote-tracking branch 'origin/product_variants_api_add_stock_to_response' into product_variants_api_add_stock_to_response

kastoras committed:
New/NEW Product Variants API, add variant stock to response

Remove dependency inserded by mistake

kastoras commented:
I have add the suggested fix.
